common/setups/rasr/util/decode.py

```diff
@@ -0,0 +1,132 @@
+__all__ = [
+    "RecognitionParameters",
+    "PriorPath",
+    "SearchJobArgs",
+    "Lattice2CtmArgs",
+    "StmArgs",
+    "ScliteScorerArgs",
+    "OptimizeJobArgs",
+]
+
+from dataclasses import dataclass
+from itertools import product
+from typing import Any, Dict, List, Optional, Tuple, TypedDict, Union
+
+from sisyphus import tk
+
+import i6_core.rasr as rasr
+
+from ..config.am_config import Tdp
+
+
+class RecognitionParameters:
+    def __init__(
+        self,
+        *,
+        am_scales: List[float],
+        lm_scales: List[float],
+        prior_scales: List[float],
+        pronunciation_scales: List[Optional[float]],
+        tdp_scales: List[float],
+        speech_tdps: List[Tdp],
+        silence_tdps: List[Tdp],
+        nonspeech_tdps: List[Optional[Tdp]],
+        altas: List[Optional[float]],
+    ):
+        self.am_scales = am_scales
+        self.lm_scales = lm_scales
+        self.prior_scales = prior_scales
+        self.pronunciation_scales = pronunciation_scales
+        self.tdp_scales = tdp_scales
+        self.speech_tdps = speech_tdps
+        self.silence_tdps = silence_tdps
+        self.nonspeech_tdps = nonspeech_tdps
+        self.altas = altas
+
+    def _get_iter(self):
+        return [
+            (am, lm, pri, pron, tdp, speech, sil, nonspeech, al)
+            for am, lm, pri, pron, tdp, speech, sil, nonspeech, al in product(
```

Comment:
I don't like that we have the cartesian product of all parameters. Sometimes I just want to have a some specific parameters to go together.. or do multiple sweeps in different directions e.g.

```
for am_scale, lm_scale in [(0.1, 1.0), (1.0, 10.0), (10., 100.0), (1.0, 5.0), (1.0, 15.0), (1.0, 20.0)]:
    do_single_recog(am_scale, lm_scale)
```

mh.. one solution would be to support `List[RecognitionParameters]` everywhere above. Or maybe define some addition operator that concatenates different cartesian products.. Or have base class and then inherit `CartesianProductRecognitionParameters` and `FlatListRecognitionParameters`..
I am open for ideas

Reply:
I added the option to also add a list of recog params. that should give you the desired fleibility. would you remove the cartesian product in general? aka two Recog Param Classes?

Reply:
I think the current implementation is fine. Thanks! 🙏